runner/views/run_api_view.py

```diff
@@ -193,26 +193,44 @@ class RunApiRestartViewSet(GenericAPIView):
     logger = logging.getLogger(__name__)
 
     def post(self, request):
-        run_id = request.data.get('run')
-        run = RunObject.from_db(run_id)
-        inputs = dict()
-        for port in run.inputs:
-            inputs[port.name] = port.db_value
-        data = dict(
-            app=str(run.run_obj.app.id),
-            inputs=inputs,
-            tags=run.tags,
-            job_group_id=run.job_group.id,
-            job_group_notifier_id=run.job_group_notifier.id,
-            resume=run_id
-        )
-        serializer = APIRunCreateSerializer(data=data, context={'request': request})
+        serializer = self.serializer_class(data=request.data)
+        if not serializer.is_valid():
+            return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+        run_id = serializer.validated_data.get('run_id')
+        group_id = serializer.validated_data.get('group_id')
+        if run_id:
+            run = get_object_or_404(Run, pk=run_id)
+            runs = RunObject.from_db(run_id)
+        elif group_id:
+            runs = [RunObject.from_db(r.id) for r in Run.objects.filter(job_group_id=group_id).all()]
```

Comment:
I don't fully understand `from_db` -- let me know if there is a better way to do this